cpp/arcticdb/util/key_utils.hpp

```diff
@@ -117,10 +117,16 @@ ankerl::unordered_dense::set<AtomKey> recurse_segment(const std::shared_ptr<stre
 inline ankerl::unordered_dense::set<AtomKey> recurse_index_key(const std::shared_ptr<stream::StreamSource>& store,
                                                      const IndexTypeKey& index_key,
                                                      const std::optional<VersionId>& version_id=std::nullopt) {
-    auto segment = store->read_sync(index_key).second;
-    auto res = recurse_segment(store, segment, version_id);
-    res.emplace(index_key);
-    return res;
+    ankerl::unordered_dense::set<AtomKey> res;
```

Comment:
The related function `recurse_index_keys` takes a `ReadOptions` on which we can set `read_opts.ignore_missing_key`. You can also pass those options down to `read_sync`. I think it's probably best to put a `ReadOptions` on this function signature.